src/timer/timer_enable.cc

```diff
@@ -427,10 +432,15 @@ TimerSection* TimerSection::SubSection(const std::string& name) {
         }
     }
 
-    std::unique_lock lock(shared_mtx_);
-    auto             insert = subsections_.emplace(
-        name,
-        std::make_unique<TimerSection>(name, collector_index_count.fetch_add(1), CtorPermission()));
+    const std::size_t collector_idx = collector_index_count.fetch_add(1);
```

Comment:
Do we need atomic when there's already mutex?

Reply:
`GetMainSection` accesses it without locks. We may be able to massage the code to protect the initialization by mutex, but we still need it for now.